doc.js

```diff
@@ -365,21 +365,21 @@ define('doc', ['event'], function(event) {
 			},
 
 			'checked' : function(value) {
-				if(!this.attr('type').match('(checkbox|radio)')) {
-					throw Error("This element is not checkbox or radio type");
-				}
-
-				if(value === null || value === undefined) {
-					return this.first().checked;
+				if(value !== undefined && typeof value !== 'boolean') {
+					throw Error("Parameter type is not valid");
 				}
 
-				if(typeof value !== 'boolean') {
-					throw Error("Parameter type is not valid");
+				if(typeof value === 'boolean') {
+					this.each( function (el) {
+						if(!el.getAttribute('type') || !el.getAttribute('type').match('(checkbox|radio)')) {
+							throw Error("This element is not checkbox or radio");
```

Comment:
Seria interessante que o erro mostrasse qual elemento selecionado foi responsável pelo erro. Também seria legal usar aspas simples para manter o padrão no código.

```suggestion
							throw Error('Element ' + el + ' is not checkbox nor radio');
```

Reply:
:hammer_and_pick: